control-plane/gateways/deployment_dataplane_container.go

```diff
@@ -186,7 +186,18 @@ func getDataplaneArgs(namespace string, config GatewayConfig, bearerTokenFile st
 		args = append(args, "-service-partition="+config.ConsulTenancyConfig.ConsulPartition)
 	}
 
-	args = append(args, "-tls-disabled")
+	tlsArgs := []string{"-tls-disabled"}
```

Comment:
This is almost long enough that it might make a good isolated function, just for clarity when reading through. Not a blocker, but it's right on the edge where I might expect a `args = append(args, tlsArgs(args, config)...)`

Reply:
honestly I was kinda thinking the same thing, gonna extract it